spot_rl_experiments/experiments/skill_test/test_view_pose_based_nav_pick_nav_place.py

```diff
@@ -0,0 +1,197 @@
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import argparse
+import os
+import os.path as osp
+import time
+
+import magnum as mn
+import numpy as np
+import rospy
+from spot_rl.envs.skill_manager import SpotSkillManager
+from spot_rl.utils.retrieve_robot_poses_from_cg import get_view_poses
+from spot_rl.utils.utils import ros_topics as rt
+from spot_rl.utils.waypoint_estimation_based_on_robot_poses_from_cg import (
+    get_navigation_points,
+)
+from spot_wrapper.utils import get_angle_between_two_vectors
+
+NUM_REPEAT = 1
+# Center, extent, object_tags
+WAYPOINT_TEST = [
+    (
+        {
+            "id": 53,
+            "bbox_extent": [0.7, 0.5, 0.5],
+            "bbox_center": [3.5, -1.2, 0.3],
+            "object_tag": "chair",
+        },
+        {
+            "id": 71,
+            "bbox_extent": [0.9, 0.9, 0.6],
+            "bbox_center": [1.1, 3.3, -0.1],
+            "object_tag": "coffee table",
+        },
+    )
+] * NUM_REPEAT
+object_to_pickup = "bottle"
+VISUALIZE = False
+PATH_PLANNING_VISUALIZATION_FOLDER = "path_planning_vis_for_cg"
+os.makedirs(PATH_PLANNING_VISUALIZATION_FOLDER, exist_ok=True)
+
+
+class SpotRosSkillExecutor:
```

Comment:
The SpotRosSkillExecutor still has duplicate code from `test_dynamicnav_pick_dynamicnav_place.py`, are we able to condense the two files?